File: app/src/main/java/i/am/shiro/amai/data/TagDao.java
package i.am.shiro.amai.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import static androidx.room.OnConflictStrategy.REPLACE;

@Dao
public interface TagDao {

    @Insert(onConflict = REPLACE)
    void insert(List<TagEntity> tagEntities);

    @Query("SELECT * FROM TagEntity WHERE bookId=:bookId AND type=:type")
    LiveData<List<TagEntity>> findByFilter(int bookId, String type);
}

File: app/src/main/java/i/am/shiro/amai/data/dao/TagDao.kt
package i.am.shiro.amai.data.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import i.am.shiro.amai.data.entity.TagEntity

@Dao
interface TagDao {

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insert(tagEntities: List<TagEntity>)
}
